components/Candidates.js

```diff
@@ -20,21 +26,58 @@ function CandidateCard({ candidate }) {
   );
 }
 
-export default function Candidates({ candidates }) {
+export default function Candidates() {
+  const {
+    status,
+    data,
+    isFetchingMore,
+    fetchMore,
+    canFetchMore,
+  } = useInfiniteQuery("candidates", fetchCandidates, {
+    getFetchMore: (lastGroup, _allGroups) => lastGroup.nextPage,
+  });
+
   return (
     <div className="mt-16">
       <h2 className="text-3xl font-extrabold">Candidatos</h2>
-      <div className="mt-10 grid grid-cols-1 md:grid-cols-3 gap-4">
-        {candidates.map((candidate) => {
-          const fullName = `${candidate.nombres}  ${candidate.apellido_paterno} ${candidate.apellido_materno}`;
-          return (
-            <CandidateCard
-              key={fullName}
-              candidate={{ ...candidate, fullName }}
-            />
-          );
-        })}
-      </div>
+      {status === "loading" ? (
+        "loading..."
+      ) : (
+        <>
+          <div className="mt-10 mb-8 grid grid-cols-1 md:grid-cols-3 gap-4">
+            {data.map((group, i) => (
+              <Fragment key={i}>
+                {group.candidates.map((candidate) => {
+                  {
+                    const fullName = `${candidate.nombres}  ${candidate.apellido_paterno} ${candidate.apellido_materno}`;
+                    return (
+                      <CandidateCard
+                        key={fullName}
+                        candidate={{ ...candidate, fullName }}
+                      />
+                    );
+                  }
+                })}
+              </Fragment>
+            ))}
+          </div>
+          <div>
+            <button
+              onClick={() => fetchMore()}
+              disabled={!canFetchMore || isFetchingMore}
+              className="text-neutral-400"
+            >
+              {isFetchingMore ? (
+                "Cargando más..."
+              ) : (
+                <span>
+                  {canFetchMore ? "Cargar más..." : "Nada más que cargar"}
```

Comment:
Can we do something to know if it's the last page and not show "Nada más que cargar"

Something like this
```diff
diff --git a/components/Candidates.js b/components/Candidates.js
index e506f43..20952fe 100644
--- a/components/Candidates.js
+++ b/components/Candidates.js
@@ -36,6 +36,7 @@ export default function Candidates() {
   } = useInfiniteQuery("candidates", fetchCandidates, {
     getFetchMore: (lastGroup, _allGroups) => lastGroup.nextPage,
   });
+  const isLastPage = data && data[data.length - 1].isLastPage;
 
   return (
     <div className="mt-16">
@@ -61,21 +62,19 @@ export default function Candidates() {
               </Fragment>
             ))}
           </div>
-          <div>
-            <button
-              onClick={() => fetchMore()}
-              disabled={!canFetchMore || isFetchingMore}
-              className="text-neutral-400"
-            >
-              {isFetchingMore ? (
-                "Cargando más..."
-              ) : (
-                <span>
-                  {canFetchMore ? "Cargar más..." : "Nada más que cargar"}
-                </span>
-              )}
-            </button>
-          </div>
+          {!isLastPage && (
+            <div>
+              <button
+                onClick={() => fetchMore()}
+                disabled={!canFetchMore || isFetchingMore}
+                className="text-neutral-400"
+              >
+                {isFetchingMore
+                  ? "Cargando más..."
+                  : canFetchMore && "Cargar más..."}
+              </button>
+            </div>
+          )}
         </>
       )}
     </div>
diff --git a/pages/api/candidates.js b/pages/api/candidates.js
index 3ace7ab..f6eed7b 100644
--- a/pages/api/candidates.js
+++ b/pages/api/candidates.js
@@ -8,8 +8,11 @@ function paginatedData(data, page = 1) {
     page * CONGRESS_LIMIT,
   );
   const nextPage = candidates.length > 0 ? page + 1 : null;
+  const isLastPage =
+    data[data.length - 1].identificador ===
+    candidates[candidates.length - 1].identificador;
 
-  return { candidates, nextPage };
+  return { candidates, nextPage, isLastPage };
 }
 
 // eslint-disable-next-line import/no-anonymous-default-export
```